machine/v1/types_nutanixprovider.go

```diff
@@ -0,0 +1,89 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NutanixMachineProviderConfig is the Schema for the nutanixmachineproviderconfigs API
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +k8s:openapi-gen=true
+type NutanixMachineProviderConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// clusterReferenceUuid is the UUID of the PE/cluster the Machine's VM will be created in
+	// +kubebuilder:validation:Required
+	ClusterReferenceUUID string `json:"clusterReferenceUuid"`
+
+	// imageReference is to identify the rhcos image uploaded to the Prism Central (PC)
+	// +kubebuilder:validation:Required
+	ImageReference ImageReference `json:"imageReference"`
+
+	// subnetUuid is the UUID of the cluster's network subnet to use for the Machine's VM
+	// +kubebuilder:validation:Required
+	SubnetUUID string `json:"subnetUuid"`
```

Comment:
Is this the only way to identify a subnet or can they be looked up, for example via tags or name?